### PR TITLE
buffers all calls till the event bus is open

### PR DESCRIPTION
**Pull request by edewit**
uses a decorator to return `ConnectableObservable` so that only when the bus is open communication will go through. This of course doesn't work when the first call is login and all calls are depended on that login call. That is why we need to make login part of this API so that the user only has to send commands

### CODE REVIEW

**Comment by vorburger** [comment]
> That is why we need to make login part of this API so that the user only has to send commands

just to clear: so we FIRST have to do that, BEFORE we can consider this PR? If so, let's do it in a new parallel PR (and then rebase this one once that one is in?) - are you on that, or did you hope this would motivate me to get on that? :smiley_cat: 

@edewit did you see this fails the build, unrelated to above I think, some TS build problem:

    ERROR in [at-loader] ./src/main/typescript/observable-wrapper.ts:17:3 
        TS1219: Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option to remove this warning.

**Comment by vorburger** [comment]
> That is why we need to make login part of this API so that the user only has to send commands

after [A Closer Look](https://www.youtube.com/playlist?list=PLJaq64dKJZoqsh7PGGUi-SARV4wUz_lVa), our `Login` command **IS** already on the `Minecraft` API interface, and no longer handled in that `ActionsConsumer`, so... what's missing here? The other remaining ones (narrate, command; registerCondition) also need to move over - but that shouldn't block this.

So.. just sort out the build issue, and this is good to go?

PS: That TS AOP with the `@buffered()` looks very fancy!

**Comment by vorburger** [comment]
> The other remaining ones (narrate, command; registerCondition) also need to move over

narrate removed in df53af5f6cf188ee1121e0b9535b7623c94cfbad, (run)command proposed to migrate in #96; registerCondition is #53 WIP.

File: scratch/tsconfig.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
     "moduleResolution": "node",
     "declaration": true,
     "sourceMap": true,
-    "emitDecoratorMetadata": false,
-    "experimentalDecorators": false,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
```
[the remark this answers is not in the document]
Comment (edewit) [reply scratch/tsconfig.json]: okay my bad that is on the wrong project

**Comment by vorburger** [comment]
merged as 2a3552322eecfdd6f7414db15e249984fe218c47 and e18cb0cf55530b5f9d42ce724fdbd0798941c264 (with fixes in 73584de693740bfe8bda58b2b1c7956e08f01df6)

**Pull request closed without merging**